### PR TITLE
[v14] e2e test redshift cluster

### PR DESCRIPTION
**Pull request by GavinFrazar**
Changelog: Fixed Redshift auto-user deactivation/deletion failure that occurs when a user is created or deleted and another user is deactivated concurrently.

Backports https://github.com/gravitational/teleport/pull/41195 and https://github.com/gravitational/teleport/pull/43335 to branch/v14.
- https://github.com/gravitational/teleport/pull/41195
- https://github.com/gravitational/teleport/pull/43335

### CODE REVIEW

**Review by greedy52 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

**Comment by GavinFrazar** [comment]
@r0mant @zmb3 needs codeowner approval since it touches the e2e test gha workflow file

**Review by r0mant — APPROVED** [review]

**Pull request merged**
